### PR TITLE
rbd: fix segmentation fault when rbd_group_image_list() getting -ENOENT

### PR DESCRIPTION
**Pull request by Songweibin**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

Fixes: http://tracker.ceph.com/issues/38468

### CODE REVIEW

**Comment by Songweibin** [comment]
@trociny  Updated, thx!

File: src/test/pybind/test_rbd.py [hunk 1]
```diff
@@ -2024,6 +2024,16 @@ def test_group_image_list(self):
         self.group.add_image(ioctx, image_name)
         eq([image_name], [img['name'] for img in self.group.list_images()])
 
+    def test_group_image_list_move_to_trash(self):
+        eq([], list(self.group.list_images()))
+        with Image(ioctx, image_name) as image:
+            image_id = image.id
```
Comment (trociny) [inline src/test/pybind/test_rbd.py]: @Songweibin  `id` is a method. It should be `image_id = image.id()`.
```
adonis:~/ceph/ceph.ci/build% ../src/vstart.sh -n
adonis:~/ceph/ceph.ci/build% RBD_FEATURES=127 PYTHONPATH=../src/test/pybind:../src/pybind:./lib/cython_modules/lib.3 LD_LIBRARY_PATH=./lib nosetests -v test_rbd:TestGroups.test_group_image_list_move_to_trash
test_rbd.TestGroups.test_group_image_list_move_to_trash ... ERROR
ERROR

======================================================================
ERROR: test_rbd.TestGroups.test_group_image_list_move_to_trash
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/mgolub/ceph/ceph.ci/src/test/pybind/test_rbd.py", line 2035, in test_group_image_list_move_to_trash
    RBD().trash_restore(ioctx, image_id, image_name)
  File "rbd.pyx", line 1377, in rbd.RBD.trash_restore
  File "rbd.pyx", line 845, in rbd.cstr
rbd.InvalidArgument: image_id must be a string
```

**Comment by Songweibin** [comment]
@trociny  Sorry for the mistake, updated. Thank you.

BTW: how can i test like you? It seems that the version is a bit low or missing a package:
```
[swb@ ~/ceph/ceph-dev/build]$ RBD_FEATURES=127 PYTHONPATH=../src/test/pybind:../src/pybind:./lib/cython_modules/lib.3 LD_LIBRARY_PATH=./lib nosetests -v test_rbd:TestGroups.test_group_image_list_move_to_trash
Failure: ImportError (/usr/lib64/python2.7/site-packages/rbd.so: undefined symbol: rbd_get_byte_rate) ... ERROR

======================================================================
ERROR: Failure: ImportError (/usr/lib64/python2.7/site-packages/rbd.so: undefined symbol: rbd_get_byte_rate)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/loader.py", line 407, in loadTestsFromName
    module = resolve_name(addr.module)
  File "/usr/lib/python2.7/site-packages/nose/util.py", line 312, in resolve_name
    module = __import__('.'.join(parts_copy))
  File "/home/swb/ceph/ceph-dev/src/test/pybind/test_rbd.py", line 15, in <module>
    from rbd import (RBD, Group, Image, ImageNotFound, InvalidArgument, ImageExists,
ImportError: /usr/lib64/python2.7/site-packages/rbd.so: undefined symbol: rbd_get_byte_rate

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (errors=1)
```

**Comment by trociny** [comment]
You are using python2, so you need to change `PYTHONPATH` to include `./lib/cython_modules/lib.2` (check that it exists in your case). Or just export the environment variables that `vstart.sh` suggests (prints) after starting the daemons. In my case the output is:
```
dashboard urls: https://10.100.2.80:41390
  w/ user/pass: admin / admin
restful urls: https://10.100.2.80:42390
  w/ user/pass: admin / 46392359-0b8f-43b6-8c7e-c8053f687579

export PYTHONPATH=./pybind:/home/mgolub/ceph/ceph.ci/src/pybind:/home/mgolub/ceph/ceph.ci/build/lib/cython_modules/lib.3::/home/mgolub/ceph/ceph.ci/src/pybind:/home/mgolub/ceph/ceph.ci/build/lib/cython_modules/lib.3
export LD_LIBRARY_PATH=/home/mgolub/ceph/ceph.ci/build/lib
CEPH_DEV=1
```

**Comment by Songweibin** [comment]
@trociny  Thank you : )  :+1:

**Pull request merged**